gno.land/cmd/gnoland/start.go

```diff
@@ -114,18 +117,42 @@ func (c *startCfg) RegisterFlags(fs *flag.FlagSet) {
 		&c.config,
 		"config",
 		"",
-		"config file (optional)",
+		"the flag config file (optional)",
+	)
+
+	fs.StringVar(
+		&c.nodeConfigPath,
+		"tm2-node-config",
+		"",
+		"the node TOML config file path (optional)",
 	)
 }
 
-func execStart(c *startCfg, args []string, io *commands.IO) error {
+func execStart(c *startCfg, io *commands.IO) error {
 	logger := log.NewTMLogger(log.NewSyncWriter(io.Out))
 	rootDir := c.rootDir
 
-	cfg := config.LoadOrMakeConfigWithOptions(rootDir, func(cfg *config.Config) {
-		cfg.Consensus.CreateEmptyBlocks = true
-		cfg.Consensus.CreateEmptyBlocksInterval = 0 * time.Second
-	})
+	var (
+		cfg        *config.Config
+		loadCfgErr error
+	)
+
+	// Set the node configuration
+	if c.nodeConfigPath != "" {
+		// Load the node configuration
+		// from the specified path
+		cfg, loadCfgErr = config.LoadConfigFile(c.nodeConfigPath)
+	} else {
+		// Load the default node configuration
+		cfg, loadCfgErr = config.LoadOrMakeConfigWithOptions(rootDir, func(cfg *config.Config) {
+			cfg.Consensus.CreateEmptyBlocks = true
+			cfg.Consensus.CreateEmptyBlocksInterval = 0 * time.Second
```

Comment:
I suggest centralizing the default configuration by creating a helper function since we have it in multiple places.

Reply:
I've looked into this, and it turns out this callback is actually useless. The default consensus configuration is creating empty blocks, at 0 interval already, so this setting has no effect.

I've updated the method to drop this support:
[fcb3e57](https://github.com/gnolang/gno/pull/1240/commits/fcb3e57bffe985b09c738dde4b7a680f50162a3f)

Reply:
We should retain the option to provide a default callback, but you can leave this one empty as it doesn't have any effect.

Reply:
Returned:

[afcb476](https://github.com/gnolang/gno/pull/1240/commits/afcb476c0f43e7ed61cb08cdef05a0556b04b605)